### ISSUE TITLE
Gvfs sparse -a fails with "dirty" git status even if "dirty" files are within sparse list

### ISSUE THREAD

**Issue by judemars** [microsoft/VFSForGit#1460]
I’ve confirmed that gvfs sparse -a seems to fail whenever your repo is dirty at all, regardless of whether or not the added/edited files are in your sparse list at all. I realize the decision to fail if the repo is dirty at all opens up some weird edge cases, however I think it is way too brittle, especially since we are talking about having tooling (git hooks and scoper) call gvfs sparse unbeknownst to the user. Would a better approach be to only fail gvfs sparse if there are dirty folders present that are outside of the sparse list you are switching to?

**Comment by judemars** [comment]
Any movement on this? I think this is a pretty big blocker for Office's adoption of gvfs sparse

**Comment by judemars** [comment]
Wow I'm impressed, this seems to work great upon first poking at it! I am at a conference this week so I probably won't have time to do corner cases until next Monday but just wanted to pop by and say thanks a lot for this and great work!

**Comment by judemars** [comment]
Well, here's an issue. What I did:

1. Starting out with otools and word folders in my sparse list, I modified a file in otools. Then,

D:\onew\src>gvfs sparse --set word
Running git status...Succeeded
Updating sparse folder set...Succeeded
Forcing a projection change...Succeeded

(At this point both otools and word appear) 

D:\onew\src>gvfs sparse --set xl
Running git status...Failed

 M otools/dirs

git status reported that you have dirty files
Either commit your changes or reset and clean
sparse was aborted

How I understood the behavior was that only calls with --prune would fail with a dirty git status, and if that's the case then I don't think the second call with xl should have failed. However, it could be that it's supposed to fail regardless of if --prune is passed if the git status is dirty, but in that case then I would have expected the first case to fail. Either they both pass or both fail makes sense, but not one or the other. But maybe I'm misunderstanding something?
